plugins/module_utils/network/sonic/config/mclag/mclag.py

```diff
@@ -193,6 +200,64 @@ def _state_deleted(self, want, have):
                     commands = update_states(diff_want, "deleted")
         return commands, requests
 
+    def _state_replaced(self, want, have, diff):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        requests = []
+        commands = []
+
+        diff1 = get_diff(have, want, TEST_KEYS, is_skeleton=True)
```

Comment:
Why is diff being re-executed here? This looks incorrect. Please use the input diff value (or explain why this is needed).

Reply:
Diff we usually get it between want and have. This diff1 is executed to get the difference between have and want. If "have" been a superset of "want".